lib/LedgerSMB/Scripts/setup.pm

```diff
@@ -775,27 +775,54 @@ verify_check => md5_hex($check->test_query),
     $sth->finish();
 
     my $heading = { map { $_ => $_ } @{$check->display_cols} };
-    my $buttons = [
+    my %buttons = map { $_ => 1 } @{$check->buttons};
+    my $buttons;
+    push @$buttons,
            { type => 'submit',
              name => 'action',
-            value => $check->columns ? 'fix_tests' : 'cancel',
-             text => $request->{_locale}->text($check->columns
-                                                ? 'Save and Retry'
-                                                : 'Cancel'),
+            value => 'fix_tests',
+          tooltip => { id => 'action-fix-tests',
+                       msg => $request->{_locale}->text($check->{tooltips}{'Save and Retry'}),
```

Comment:
Please write `$check->{tooltips}->{'Save and Retry'}` instead; it's the style we use throughout the code base (note, there are other occurrances of the same below).